src/CatalogGraphQLClient.php

```diff
@@ -108,6 +197,24 @@ public function find(array $searchBody, array $fields, int $page = 1, $perPage =
      * @throws NonExistingFieldException
      */
     public function findOne(array $searchBody, array $fields): ?Product
+    {
+        $result = $this->findOneItem(Types\Query::PRODUCT, $searchBody, $fields);
+
+        if ($result === null) {
+            return null;
+        }
+
+        return new Product($result);
```

Comment:
Same here, just make it a proxy
```suggestion
        return $this->findOneProduct($searchBody, $fields);
```